mhs/mhs/common/state/tests/test_dynamo_persistence_adaptor.py

```diff
@@ -30,146 +28,213 @@
 
 
 class TestDynamoPersistenceAdaptor(unittest.TestCase):
+    """Unit tests for the DynamoPersistenceAdaptor"""
 
-    @patch.object(dynamo_persistence_adaptor.aioboto3, "resource")
-    def setUp(self, mock_boto3_resource):
-        self.mock_dynamodb = mock_boto3_resource.return_value
-        self.mock_table = self.mock_dynamodb.Table.return_value
+    def setUp(self):
+        """Configure a dynamo persistence adaptor with the boto3 calls mocked."""
 
         self.service = DynamoPersistenceAdaptor(
-            state_table=TEST_STATE_TABLE,
-            state_arn=TEST_STATE_ARN,
-            sync_async_table=TEST_ASYNC_TABLE,
-            sync_async_arn=TEST_ASYNC_ARN
+            table_name=TEST_TABLE_ARN
         )
 
-    #   TESTING ADD METHOD
-    @async_test
-    async def test_add_success_state(self):
-        self.__setFutureResponse(self.mock_table.put_item, TEST_ADD_EMPTY_RESPONSE)
-
-        response = await self.service.add(TEST_STATE_TABLE, TEST_KEY, TEST_ITEM)
-
-        self.assertIsNone(response)
-
+    @patch.object(dynamo_persistence_adaptor.aioboto3, "resource")
```

Comment:
I believe that this decorator can be used as a class decorator instead of having to repeat it for each test in this class. Actually, you're also repeating `self.__configure_mocks` in all of these tests, so maybe instead don't do it as a class decorator, and do something like:
```python
def setUp(self):
    ...
    patcher = patch.object(...)
    self.mock = patcher.start()
    self.addCleanup(patcher.stop)
    self.__configure_mocks(mock)
```

Reply:
done.